test/helpers/position_assertions.cc

```diff
@@ -1129,4 +1130,119 @@ void ApplyCodeActionAssertion::check(const UnorderedMap<std::string, std::shared
+                                                              string_view assertionType) {
+    static const regex contentsRegex(R"(^\s*\"([^\"]+)\"\s*$)");
+
+    smatch matches;
+    string assertionContentsString = string(assertionContents);
+    if (regex_search(assertionContentsString, matches, contentsRegex)) {
+        auto query = matches[1].str();
+        return make_shared<SymbolSearchAssertion>(filename, range, assertionLine, query);
+    }
+
+    ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
+        "Improperly formatted assertion. Expected 'symbol-search: \"<query>\"'. Found '{}'", assertionContents);
+
+    return nullptr;
+}
+
+void SymbolSearchAssertion::checkAllForQuery(std::string query,
+                                             const vector<shared_ptr<SymbolSearchAssertion>> &assertions,
+                                             const UnorderedMap<string, shared_ptr<core::File>> &sourceFileContents,
+                                             LSPWrapper &lspWrapper, int &nextId, string_view uriPrefix,
+                                             string errorPrefix) {
+    const int id = nextId++;
+    const int lineWidth = 6; // ensure sort order == numerical order (if file is less than a million lines)
+
+    std::map<std::string, vector<shared_ptr<SymbolSearchAssertion>>> assertionsByLine;
+    for (auto assertion : assertions) {
+        auto location = assertion->getLocation(uriPrefix);
+        auto line = fmt::format("{}:{:{}}", location->uri, location->range->start->line, lineWidth);
+        assertionsByLine[line].push_back(assertion);
+    }
+
+    // Request results from LSP
+    auto responses = lspWrapper.getLSPResponsesFor(makeWorkspaceSymbolRequest(id, query));
+    ASSERT_EQ(1, responses.size()) << "Unexpected number of responses to a `textDocument/definition` request.";
+    ASSERT_NO_FATAL_FAILURE(assertResponseMessage(id, *responses.at(0)));
+    auto &respMsg = responses.at(0)->asResponse();
+    ASSERT_TRUE(respMsg.result.has_value());
+    ASSERT_FALSE(respMsg.error.has_value());
+    auto &result = *(respMsg.result);
+    vector<shared_ptr<SymbolInformation>> symbols;
+    if (auto symbolInfos = get_if<vector<unique_ptr<SymbolInformation>>>(
+            &get<variant<JSONNullObject, vector<unique_ptr<SymbolInformation>>>>(result))) {
+        for (auto &symbolInfo : *symbolInfos) {
+            symbols.push_back(shared_ptr<SymbolInformation>(std::move(symbolInfo)));
+        }
+    }
+    std::map<std::string, vector<shared_ptr<SymbolInformation>>> symbolsByLine;
+    for (auto &symbol : symbols) {
+        auto &location = symbol->location;
+        auto line = fmt::format("{}:{:{}}", location->uri, location->range->start->line, lineWidth);
+        symbolsByLine[line].push_back(symbol);
+    }
+
+    bool success = true;
+    // Show any expected symbols that weren't returned by LSP
```

Comment:
Note that this test is checking that Sorbet returns the expected locations of search results, but doesn't check the `name`/`containerName` associated with each result. idk if that's something you think is important to test or is worth further exploration in the future.

Reply:
I agree that it makes sense to check these things, but it's probably tedious to have to specify all of them, every single time, so my expectation is that the next step here is to expand SymbolSearchAssertion to have a bunch of optional fields that check `name`, `containerName`, exact `location.range` etc.?

Reply:
Works for me. I just wanted to raise the issue in case you hadn't thought about it.